onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -6583,10 +6583,11 @@ def aten_prelu_backward(
     raise NotImplementedError()
 
 
-def aten_prod(self: TensorType, dtype: Optional[int] = None) -> TensorType:
+@torch_op(("aten::prod", "aten::prod.dim_int"), trace_only=True)
+def aten_prod(self: TReal, dim: int, keepdim: bool = False) -> TReal:
     """prod(Tensor self, *, ScalarType? dtype=None) -> Tensor"""
 
-    raise NotImplementedError()
+    return op.ReduceProd(self, axes=[dim], keepdims=keepdim)
```

Comment:
Are tests available?

Reply:
Added 'Todo: add test for this function later'. Will do that after finish all the backward functions.